cpp/src/arrow/flight/sql/server_test.cc

```diff
@@ -381,6 +382,49 @@ TEST_F(TestFlightSqlServer, TestCommandGetTablesWithIncludedSchemas) {
   AssertTablesEqual(*expected_table, *table);
 }
 
+TEST_F(TestFlightSqlServer, TestCommandGetTypeInfo) {
+  ASSERT_OK_AND_ASSIGN(auto flight_info, sql_client->GetTypeInfo({}));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  const std::shared_ptr<Schema>& expected_schema = SqlSchema::GetTypeInfoSchema();
+
+  const std::shared_ptr<RecordBatch>& batch =
+      example::DoGetTypeInfoResult(expected_schema);
+
+  ASSERT_OK_AND_ASSIGN(auto expected_table, Table::FromRecordBatches({batch}));
+  std::shared_ptr<Table> table;
+  ASSERT_OK(stream->ReadAll(&table));
+
+  AssertTablesEqual(*expected_table, *table);
+}
+
+TEST_F(TestFlightSqlServer, TestCommandGetTypeInfoWithFiltering) {
+  int data_type = -4;
+  ASSERT_OK_AND_ASSIGN(auto flight_info, sql_client->GetTypeInfo({}, data_type));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  const std::shared_ptr<Schema>& expected_schema = SqlSchema::GetTypeInfoSchema();
+
+  const std::shared_ptr<RecordBatch>& batch =
+      example::DoGetTypeInfoResult(expected_schema, data_type);
+
+  const arrow::Result<std::shared_ptr<Table>>& expected_table_result =
```

Comment:
Also note that this test needs to be updated like the above one was.

Reply:
Fixed